### ISSUE TITLE
CentOS 7 can't upgrade cockpit-networkmanager-169 -> cockpit-system-176

### ISSUE THREAD

**Issue by kai4785** [cockpit-project/cockpit#11905]
Cockpit version: 176
OS: CentOS 7
Page: N/A

When using 'yum' to update from cockpit 169 to 176, I run into file conflicts.
```
$ sudo yum update cockpit\*
Loaded plugins: auto-update-debuginfo, fastestmirror, langpacks, nvidia
Loading mirror speeds from cached hostfile
Resolving Dependencies
--> Running transaction check
---> Package cockpit.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit.x86_64 0:176-4.el7.centos will be an update
---> Package cockpit-bridge.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-bridge.x86_64 0:176-4.el7.centos will be an update
---> Package cockpit-dashboard.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-dashboard.x86_64 0:176-4.el7.centos will be an update
---> Package cockpit-machines.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-machines.x86_64 0:176-4.el7.centos will be an update
---> Package cockpit-packagekit.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-packagekit.x86_64 0:176-4.el7.centos will be an update
---> Package cockpit-storaged.noarch 0:169-1.el7.centos will be updated
---> Package cockpit-storaged.noarch 0:176-4.el7.centos will be an update
---> Package cockpit-system.noarch 0:169-1.el7.centos will be updated
---> Package cockpit-system.noarch 0:176-4.el7.centos will be an update
---> Package cockpit-ws.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-ws.x86_64 0:176-4.el7.centos will be an update
--> Finished Dependency Resolution

Dependencies Resolved

===================================================================================================================================================================================================================
 Package                                                Arch                                       Version                                                 Repository                                         Size
===================================================================================================================================================================================================================
Updating:
 cockpit                                                x86_64                                     176-4.el7.centos                                        0cache-extras                                      47 k
 cockpit-bridge                                         x86_64                                     176-4.el7.centos                                        0cache-extras                                     497 k
 cockpit-dashboard                                      x86_64                                     176-4.el7.centos                                        0cache-extras                                     203 k
 cockpit-machines                                       x86_64                                     176-4.el7.centos                                        0cache-extras                                     518 k
 cockpit-packagekit                                     x86_64                                     176-4.el7.centos                                        0cache-extras                                     222 k
 cockpit-storaged                                       noarch                                     176-4.el7.centos                                        0cache-extras                                     348 k
 cockpit-system                                         noarch                                     176-4.el7.centos                                        0cache-extras                                     1.2 M
 cockpit-ws                                             x86_64                                     176-4.el7.centos                                        0cache-extras                                     796 k

Transaction Summary
===================================================================================================================================================================================================================
Upgrade  8 Packages

Total size: 3.8 M
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test


Transaction check error:
  file /usr/share/cockpit/networkmanager/manifest.json from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/index.html.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.ca.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.cs.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.de.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.es.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.eu.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.fi.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.fr.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.hr.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.hu.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.ja.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.ko.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.my.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.nl.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.pa.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.pl.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.pt.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.pt_BR.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.tr.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.uk.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/po.zh_CN.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/firewall.css.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/firewall.min.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch
  file /usr/share/cockpit/networkmanager/network.min.js.gz from install of cockpit-system-176-4.el7.centos.noarch conflicts with file from package cockpit-networkmanager-169-1.el7.centos.noarch

Error Summary
-------------

```

Steps to reproduce

1. Install cockpit-networkmanager-169 (old)
2. yum update cockpit\*

The fix is relatively simple, but I don't fully understand why. Simply remove the version portion of the obsolete:
https://github.com/cockpit-project/cockpit/blob/master/tools/cockpit.spec#L472

I don't know why the version number is there or if it's needed, but with out it, it works fine.
```
$ sudo yum update ../RPMS/*/cockpit*.rpm ../RPMS/*/cockpit-bridge*.rpm ../RPMS/*/cockpit-dashboard*.rpm ../RPMS/*/cockpit-system*.rpm ../RPMS/*/cockpit-machines*.rpm ../RPMS/*/cockpit-storaged*.rpm ../RPMS/*/cockpit-ws*.rpm ../RPMS/*/cockpit-networkmanager*.rpm ../RPMS/*/cockpit-packagekit*.rpm
Loaded plugins: auto-update-debuginfo, fastestmirror, langpacks, nvidia
Examining ../RPMS/noarch/cockpit-machines-ovirt-176-4.el7.noarch.rpm: cockpit-machines-ovirt-176-4.el7.noarch
Package cockpit-machines-ovirt not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/noarch/cockpit-storaged-176-4.el7.noarch.rpm: cockpit-storaged-176-4.el7.noarch
Marking ../RPMS/noarch/cockpit-storaged-176-4.el7.noarch.rpm as an update to cockpit-storaged-169-1.el7.centos.noarch
Examining ../RPMS/noarch/cockpit-system-176-4.el7.noarch.rpm: cockpit-system-176-4.el7.noarch
Marking ../RPMS/noarch/cockpit-system-176-4.el7.noarch.rpm as an update to cockpit-system-169-1.el7.centos.noarch
Examining ../RPMS/x86_64/cockpit-176-4.el7.x86_64.rpm: cockpit-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-176-4.el7.x86_64.rpm as an update to cockpit-169-1.el7.centos.x86_64
Examining ../RPMS/x86_64/cockpit-bridge-176-4.el7.x86_64.rpm: cockpit-bridge-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-bridge-176-4.el7.x86_64.rpm as an update to cockpit-bridge-169-1.el7.centos.x86_64
Examining ../RPMS/x86_64/cockpit-dashboard-176-4.el7.x86_64.rpm: cockpit-dashboard-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-dashboard-176-4.el7.x86_64.rpm as an update to cockpit-dashboard-169-1.el7.centos.x86_64
Examining ../RPMS/x86_64/cockpit-debuginfo-176-4.el7.x86_64.rpm: cockpit-debuginfo-176-4.el7.x86_64
Package cockpit-debuginfo not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/x86_64/cockpit-doc-176-4.el7.x86_64.rpm: cockpit-doc-176-4.el7.x86_64
Package cockpit-doc not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/x86_64/cockpit-docker-176-4.el7.x86_64.rpm: cockpit-docker-176-4.el7.x86_64
Package cockpit-docker not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/x86_64/cockpit-kubernetes-176-4.el7.x86_64.rpm: cockpit-kubernetes-176-4.el7.x86_64
Package cockpit-kubernetes not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/x86_64/cockpit-machines-176-4.el7.x86_64.rpm: cockpit-machines-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-machines-176-4.el7.x86_64.rpm as an update to cockpit-machines-169-1.el7.centos.x86_64
Examining ../RPMS/x86_64/cockpit-packagekit-176-4.el7.x86_64.rpm: cockpit-packagekit-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-packagekit-176-4.el7.x86_64.rpm as an update to cockpit-packagekit-169-1.el7.centos.x86_64
Examining ../RPMS/x86_64/cockpit-pcp-176-4.el7.x86_64.rpm: cockpit-pcp-176-4.el7.x86_64
Package cockpit-pcp not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/x86_64/cockpit-tests-176-4.el7.x86_64.rpm: cockpit-tests-176-4.el7.x86_64
Package cockpit-tests not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/x86_64/cockpit-ws-176-4.el7.x86_64.rpm: cockpit-ws-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-ws-176-4.el7.x86_64.rpm as an update to cockpit-ws-169-1.el7.centos.x86_64
Examining ../RPMS/x86_64/cockpit-bridge-176-4.el7.x86_64.rpm: cockpit-bridge-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-bridge-176-4.el7.x86_64.rpm as an update to cockpit-bridge-169-1.el7.centos.x86_64
Examining ../RPMS/x86_64/cockpit-dashboard-176-4.el7.x86_64.rpm: cockpit-dashboard-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-dashboard-176-4.el7.x86_64.rpm as an update to cockpit-dashboard-169-1.el7.centos.x86_64
Examining ../RPMS/noarch/cockpit-system-176-4.el7.noarch.rpm: cockpit-system-176-4.el7.noarch
Marking ../RPMS/noarch/cockpit-system-176-4.el7.noarch.rpm as an update to cockpit-system-169-1.el7.centos.noarch
Examining ../RPMS/noarch/cockpit-machines-ovirt-176-4.el7.noarch.rpm: cockpit-machines-ovirt-176-4.el7.noarch
Package cockpit-machines-ovirt not installed, cannot update it. Run yum install to install it instead.
Examining ../RPMS/x86_64/cockpit-machines-176-4.el7.x86_64.rpm: cockpit-machines-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-machines-176-4.el7.x86_64.rpm as an update to cockpit-machines-169-1.el7.centos.x86_64
Examining ../RPMS/noarch/cockpit-storaged-176-4.el7.noarch.rpm: cockpit-storaged-176-4.el7.noarch
Marking ../RPMS/noarch/cockpit-storaged-176-4.el7.noarch.rpm as an update to cockpit-storaged-169-1.el7.centos.noarch
Examining ../RPMS/x86_64/cockpit-ws-176-4.el7.x86_64.rpm: cockpit-ws-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-ws-176-4.el7.x86_64.rpm as an update to cockpit-ws-169-1.el7.centos.x86_64
Loading mirror speeds from cached hostfile
No Match for argument: ../RPMS/*/cockpit-networkmanager*.rpm
No package ../RPMS/*/cockpit-networkmanager*.rpm available.
Examining ../RPMS/x86_64/cockpit-packagekit-176-4.el7.x86_64.rpm: cockpit-packagekit-176-4.el7.x86_64
Marking ../RPMS/x86_64/cockpit-packagekit-176-4.el7.x86_64.rpm as an update to cockpit-packagekit-169-1.el7.centos.x86_64
Resolving Dependencies
--> Running transaction check
---> Package cockpit.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit.x86_64 0:176-4.el7 will be an update
---> Package cockpit-bridge.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-bridge.x86_64 0:176-4.el7 will be an update
---> Package cockpit-dashboard.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-dashboard.x86_64 0:176-4.el7 will be an update
---> Package cockpit-machines.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-machines.x86_64 0:176-4.el7 will be an update
---> Package cockpit-networkmanager.noarch 0:169-1.el7.centos will be obsoleted
---> Package cockpit-packagekit.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-packagekit.x86_64 0:176-4.el7 will be an update
---> Package cockpit-storaged.noarch 0:169-1.el7.centos will be updated
---> Package cockpit-storaged.noarch 0:176-4.el7 will be an update
---> Package cockpit-system.noarch 0:169-1.el7.centos will be updated
---> Package cockpit-system.noarch 0:176-4.el7 will be obsoleting
---> Package cockpit-ws.x86_64 0:169-1.el7.centos will be updated
---> Package cockpit-ws.x86_64 0:176-4.el7 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

===================================================================================================================================================================================================================
 Package                                            Arch                                   Version                                      Repository                                                            Size
===================================================================================================================================================================================================================
Installing:
 cockpit-system                                     noarch                                 176-4.el7                                    /cockpit-system-176-4.el7.noarch                                     1.2 M
     replacing  cockpit-networkmanager.noarch 169-1.el7.centos
Updating:
 cockpit                                            x86_64                                 176-4.el7                                    /cockpit-176-4.el7.x86_64                                             47 k
 cockpit-bridge                                     x86_64                                 176-4.el7                                    /cockpit-bridge-176-4.el7.x86_64                                     802 k
 cockpit-dashboard                                  x86_64                                 176-4.el7                                    /cockpit-dashboard-176-4.el7.x86_64                                  246 k
 cockpit-machines                                   x86_64                                 176-4.el7                                    /cockpit-machines-176-4.el7.x86_64                                   490 k
 cockpit-packagekit                                 x86_64                                 176-4.el7                                    /cockpit-packagekit-176-4.el7.x86_64                                 193 k
 cockpit-storaged                                   noarch                                 176-4.el7                                    /cockpit-storaged-176-4.el7.noarch                                   317 k
 cockpit-ws                                         x86_64                                 176-4.el7                                    /cockpit-ws-176-4.el7.x86_64                                         1.5 M

Transaction Summary
===================================================================================================================================================================================================================
Install  1 Package
Upgrade  7 Packages

Total size: 4.8 M
Is this ok [y/d/N]: 
```

**Comment by martinpitt** [comment]
Thanks for investigating! Fixed in PR #11906 .

**Comment by TrevorPH** [comment]
There are also missing Obsoletes: for cockpit-kdump.noarch cockpit-sosreport.noarch cockpit-networkmanager.noarch cockpit-selinux.noarch all of which now appear to be included in cockpit-system
